sound/soc/soc-pcm.c

```diff
@@ -2502,18 +2502,13 @@ static int dpcm_fe_dai_prepare(struct snd_pcm_substream *substream)
 
 	dpcm_set_fe_update_state(fe, stream, SND_SOC_DPCM_UPDATE_FE);
 
-	/* there is no point preparing this FE if there are no BEs */
-	if (list_empty(&fe->dpcm[stream].be_clients)) {
-		dev_err(fe->dev, "ASoC: no backend DAIs enabled for %s\n",
```

Comment:
no, we should not remove this. this is a key issue on the SST/Atom driver and the only way to catch when the mixer configurations are incorrect and no ability to play somewhere.

Reply:
@plbossart, hmm ok. I guess then only choice I see is to try to do this with virtual widget end point. Or some other means of having the BE existing. Anyway I see it the right way because the branched pipeline should have a scheduler, it just means changes in the firmware. I'll see what I can do.